src/webgl-surface/util/quad-tree.ts

```diff
@@ -433,7 +438,39 @@ export class Node<T extends Bounds<any>> {
   queryPoint(p: any, list: T[], visit?: IVisitFunction<T>): T[] {
     this.children.forEach((c, index) => {
       if (c.containsPoint(p)) {
-        list.push(c);
+        if (c instanceof CurvedLineShape) {
```

Comment:
This should not be here. You should override containsPoint within CurvedLineShape. Otherwise you will make ALL quad tree usages depend on this code which degrades performance for other implementations.